modules/syslog.lua

```diff
@@ -30,9 +30,11 @@ local msg_any         = octet^0
 local msg_utf8        = bom * utf8_string
 local msg             = msg_utf8 + msg_any
 local hostname        = nilvalue + printusascii^-255
-local sd_name         = printusascii^-32
+local sd_name         = (printusascii - l.S'=]" ')^-32
 local param_name      = sd_name
-local param_value     = utf8_string
+local esc             = l.P'\\'
+local param_value_esc = l.S'"\\]'
+local param_value     = ((octet - param_value_esc) + (esc * param_value_esc))^0
```

Comment:
The param_value parser would incorrectly fail on something like this: item="foo\bar"
http://tools.ietf.org/html/rfc5424#section-6.3.3
`   A backslash ('\') followed by none of the three described characters
   is considered an invalid escape sequence.  In this case, the
   backslash MUST be treated as a regular backslash and the following
   character as a regular character.  Thus, the invalid sequence MUST
   not be altered.`